src/modules/auth/auth.module.ts

```diff
@@ -68,6 +71,17 @@ export class AuthModule implements NestModule {
         { path: '/v1/login/siwe/initiate', method: RequestMethod.POST },
         { path: '/v1/login/siwe/verify', method: RequestMethod.POST }
       )
-      .forRoutes({ path: '/*', method: RequestMethod.ALL });
+      .forRoutes({ path: '/*', method: RequestMethod.ALL })
+      .apply((_req: Request, res: Response, next: NextFunction) => {
+        if (!this.configService.get('BLOCKNUM_AUTH_ENABLED')) {
+          res
+            .status(404)
+            .send(
+              'Authentication at this endpoint is disabled. Other authentication protocols may be available'
+            );
+        }
+        next();
```

Comment:
@JGiter when testing against a ssi-hub running locally on my machine, I'm seeing a possible issue

Repro steps:
1. Run ssi-hub locally with `BLOCKNUM_AUTH_ENABLED=false`
2. Try to authenticate against `/login` endpoint (e.g. with ICL < v8)

Observed result:
Following error is logged in ssi-hub
```
[Nest] 90161  - 03/01/2023, 3:15:22 PM   ERROR [ExceptionsHandler] Cannot set headers after they are sent to the client
Error: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:371:5)
    at ServerResponse.setHeader (node:_http_outgoing:576:11)
    at ServerResponse.header (/Users/john/ew/ssi-hub/node_modules/express/lib/response.js:794:10)
    at ServerResponse.append (/Users/john/ew/ssi-hub/node_modules/express/lib/response.js:755:15)
    at ServerResponse.res.cookie (/Users/john/ew/ssi-hub/node_modules/express/lib/response.js:884:8)
    at LoginController.login (/modules/auth/login.controller.ts:73:9)
```
To me, the above suggests that maybe `login` method of `login.controller` is still being executed after response is being sent.

Maybe the issue is with this `next()` call

Reply:
Yes, after removing `next`, error is not logged in ssi-hub
```suggestion
```

Reply:
may be `next()` is not needed at all. I'll check this issue. thanks to catching this

Reply:
the issue was not caught becasue there is no straight way to change environment variables from test. We need to figure out how this can be improved